### PR TITLE
Added in the ability to revert the deployment for stage and prod by a…

### PR DESCRIPTION
**Pull request by Dillon-Dumesnil**
…ctivating the respective pipeline

@edx/pipeline-team Please review. @clintonb FYI

### CODE REVIEW

File: edxpipelines/pipelines/deploy_marketing_site.py [hunk 1]
```diff
@@ -62,48 +64,80 @@ def install_pipeline(save_config_locally, dry_run, variable_files, cmd_line_vars
...
+             BuildArtifact('target/prod_tag_name.txt')])
+    )
+
+    # Stage to create and push a tag to Acquia.
     push_to_acquia_stage = pipeline.ensure_stage(PUSH_TO_ACQUIA_STAGE_NAME)
     push_to_acquia_job = push_to_acquia_stage.ensure_job(PUSH_TO_ACQUIA_JOB_NAME)
     # Ensures the tag name is accessible in future jobs.
     push_to_acquia_job.ensure_artifacts(
-        set([BuildArtifact('target/tag_name.txt')])
+        set([BuildArtifact('target/new_tag_name.txt')])
```
Comment (macdiesel) [inline edxpipelines/pipelines/deploy_marketing_site.py]: you use new_tag_name 10 times in the code below. Lets make it a constant.

File: edxpipelines/pipelines/rollback_stage_marketing_site.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import click
+from gomatic import *
+
+# Used to import edxpipelines files - since the module is not installed.
+sys.path.append(path.dirname(path.dirname(path.dirname(path.abspath(__file__)))))
+
+import edxpipelines.utils as utils
+import edxpipelines.patterns.tasks as tasks
+
+DEPLOY_PIPELINE_NAME = 'deploy-marketing-site'
```
Comment (macdiesel) [inline edxpipelines/pipelines/rollback_stage_marketing_site.py]: DEPLOY_MARKETING_PIPELINE_NAME OR DEPLOY_MKTG_PIPELINE_NAME

File: edxpipelines/pipelines/rollback_prod_marketing_site.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import click
+from gomatic import *
+
+# Used to import edxpipelines files - since the module is not installed.
+sys.path.append(path.dirname(path.dirname(path.dirname(path.abspath(__file__)))))
+
+import edxpipelines.utils as utils
+import edxpipelines.patterns.tasks as tasks
+
+DEPLOY_PIPELINE_NAME = 'deploy-marketing-site'
```
Comment (macdiesel) [inline edxpipelines/pipelines/rollback_prod_marketing_site.py]: DEPLOY_MARKETING_PIPELINE_NAME OR DEPLOY_MKTG_PIPELINE_NAME

File: edxpipelines/pipelines/rollback_stage_marketing_site.py (hunk 2 above)
Comment (macdiesel) [inline edxpipelines/pipelines/rollback_stage_marketing_site.py]: Also you use this in 2 different files, put it in constants.

Comment (macdiesel) [inline edxpipelines/pipelines/rollback_stage_marketing_site.py]: would also suggest using that constant to name the marketing deployment pipeline

File: edxpipelines/pipelines/deploy_marketing_site.py [hunk 4]
```diff
@@ -62,48 +64,80 @@ def install_pipeline(save_config_locally, dry_run, variable_files, cmd_line_vars
...
         )
     )
 
+    # Set up Acquia Github key for use in pushing tag to Acquia
+    push_to_acquia_job.add_task(
+        ExecTask(
+            [
+                '/bin/bash',
+                '-c',
+                'touch acquia_github_key.pem && '
```
Comment (macdiesel) [inline edxpipelines/pipelines/deploy_marketing_site.py]: Could you make this in to a format RSA key task please?

File: edxpipelines/pipelines/rollback_prod_marketing_site.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    config = utils.merge_files_and_dicts(variable_files, list(cmd_line_vars, ))
+
+    configurator = GoCdConfigurator(
+        HostRestClient(config['gocd_url'], config['gocd_username'], config['gocd_password'], ssl=True))
+
+    pipeline = configurator \
+        .ensure_pipeline_group('Deploy') \
+        .ensure_replacement_of_pipeline('rollback-prod-marketing-site') \
+        .set_git_material(GitMaterial('https://github.com/edx/tubular', polling=False, destination_directory='tubular')) \
+        .ensure_material(PipelineMaterial(DEPLOY_PIPELINE_NAME, FETCH_TAG_STAGE_NAME))
```
Comment (macdiesel) [inline edxpipelines/pipelines/rollback_prod_marketing_site.py]: FETCH_TAG_STAGE_NAME is also used in a few scripts, please move to constants.

**Comment by macdiesel** [comment]
@Dillon-Dumesnil good work! A few comments but it looks good. When you make the next set of changes please do not squash them and just push the new commit. This will make it easier to review the changes. You can then squash everything together after the reviews are done.

**Comment by Dillon-Dumesnil** [comment]
@macdiesel Updated with new commit.

**Comment by macdiesel** [comment]
👍

**Comment by Dillon-Dumesnil** [comment]
@macdiesel Made a very minor change with the tag's release name. Added in hours and minutes to support multiple releases in a day. Merge is on you

**Pull request merged**